services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3320,6 +3308,49 @@ public void setStorePartitionCount(String clusterName, String storeName, int par
     });
   }
 
+  void preCheckStorePartitionCountUpdate(String clusterName, Store store, int newPartitionCount) {
+    String errorMessagePrefix = "Store update error for " + store.getName() + " in cluster: " + clusterName + ": ";
+    VeniceControllerClusterConfig clusterConfig = getHelixVeniceClusterResources(clusterName).getConfig();
+    if (store.isHybrid() && store.getPartitionCount() != newPartitionCount) {
+      // Allow the update if partition count is not configured and the new partition count matches RT partition count
+      if (store.getPartitionCount() == 0) {
+        TopicManager topicManager;
+        if (isParent()) {
+          // RT might not exist in parent colo. Get RT partition count from a child colo.
+          String childDatacenter = clusterConfig.getChildDatacenters().split(",\\s*")[0];
```

Comment:
Seems that there is a util function `parseCommaSeparatedStringToSet` in Utils package doing a similar task, can we call it from here?

Reply:
Thanks for suggesting. `parseCommaSeparatedStringToList` splits by `","`. Will update it to `",\\s*"` and then use it here.